### PR TITLE
Set extended default advertising to be 1285ms

### LINKED ISSUE #32275: [TCR] Move to updated guidance in spec for extended advertisement
Extended advertisement is being recommended at 1285ms in https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8965.

Once the spec PR merges, the SDK must be updated to prefer 1285ms in extended mode.

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
- 1285ms is more reliable than 1200ms in general (https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8965)
- Update TC-DD-2.1 script to match (https://github.com/CHIP-Specifications/chip-test-plans/pull/4020)
- Update EFR32 platform to be tolerant of min==max for advertising rate

Issue #32274
Fixes #32275

Testing done:

- Linux builds, still discoverable, extended rate 1285ms
  - Tested with nRFConnect
- EFR32 builds, still discoverable, extended rate 1285ms +/- 2ms
  - Tested with nRFConnect and xG24 Explorer kit

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_DD_2_1.yaml [hunk 1]
```diff
@@ -123,8 +123,9 @@ tests:
           Try NRF Connect app (https://www.nordicsemi.com/Products/Development-tools/nrf-connect-for-desktop)
           OR
           HCIDump (https://ubuntu.com/core/docs/bluez/reference/commands)
-          ->For T0 and 30s we have to get advertisement range between 20ms to 60ms
-          ->For 30s and 15mins we have to get advertisement range between 150ms to 1200ms
+          ->Between [0 .. 30s[ expect advertisement range between 20ms to 60ms.
```
Comment (tcarmelveilleux) [inline src/app/tests/suites/certification/Test_TC_DD_2_1.yaml:126]: Merge gate comment: resolve only after https://github.com/CHIP-Specifications/chip-test-plans/pull/4020 is merged.

File: src/include/platform/CHIPDeviceConfig.h [hunk 2]
```diff
@@ -647,28 +647,26 @@
...
  * The minimum interval (in units of 0.625ms) at which the device will send BLE advertisements while
- * in the extended advertising mode. The minimum interval shall not be smaller than the default value
- * and should not be equal to the CHIP_DEVICE_CONFIG_BLE_EXT_ADVERTISING_INTERVAL_MAX.
+ * in the extended advertising mode. The minimum interval shall not be smaller than the default value.
  *
- * Defaults to 1920 (1200 ms).
+ * Defaults to 2056 (1285 ms).
  */
-#define CHIP_DEVICE_CONFIG_BLE_EXT_ADVERTISING_INTERVAL_MIN 1920
+#define CHIP_DEVICE_CONFIG_BLE_EXT_ADVERTISING_INTERVAL_MIN 2056
```
Comment (tcarmelveilleux) [inline src/include/platform/CHIPDeviceConfig.h:654]: Merge gate comment: merge only after https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/8965 is merged.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by mkardous-silabs — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DD_2_1.yaml (hunk 1 above)
Comment (robszewczyk) [reply src/app/tests/suites/certification/Test_TC_DD_2_1.yaml:126]: Spec PR has been merged, with broad review / approval

**Review by robszewczyk — APPROVED** [review]

File: src/include/platform/CHIPDeviceConfig.h (hunk 2 above)
Comment (robszewczyk) [reply src/include/platform/CHIPDeviceConfig.h:654]: Referenced pull request has been merged.

File: src/app/tests/suites/certification/Test_TC_DD_2_1.yaml (hunk 1 above)
Comment (robszewczyk) [reply src/app/tests/suites/certification/Test_TC_DD_2_1.yaml:126]: Also merged

**Review by yunhanw-google — APPROVED** [review]

**Pull request merged**